clinica/iotools/converters/genfi_to_bids/genfi_to_bids_utils.py

```diff
@@ -626,3 +788,26 @@ def correct_fieldmaps_name(to: PathLike) -> None:
 
     for z in Path(to).rglob("*phasediff_e*_ph*"):
         os.rename(z, z.parent / re.sub(r"phasediff_e[1-9]_ph", "phasediff", z.name))
+
+
+def merge_philips_diffusion(
+    to: PathLike,
+    number_of_parts: float,
+    run_num: str,
+    bids_full_path: str,
+) -> None:
+    """Adds the dwi number for each run of Philips images."""
+    import json
+
+    json_path = bids_full_path + ".json"
+    with open(to / str(json_path), "r+") as f:
+        json_file = json.load(f)
+        if int(number_of_parts) == 9:
+            if run_num in (f"run-0{k}" for k in range(1, 5)):
+                json_file["MultipartID"] = "dwi_2"
+            if run_num in (f"run-0{k}" for k in range(5, 10)):
+                json_file["MultipartID"] = "dwi_1"
+        if int(number_of_parts) == 5:
+            if run_num in (f"run-0{k}" for k in range(1, 5)):
+                json_file["MultipartID"] = "dwi_1"
+        json.dump(json_file, f, indent=4)
```

Comment:
OK, given what you told me yesterday, I think the enum should have a third variant representing any number of parts that is not 5 or 9 which are the cases for which we can do something:

```python
from enum import Enum

class PhillipsNumberOfParts(Enum):
    """DWI scans obtained with a Phillips scanner might have
    been divided in either 5 or 9 parts. This distinction is important
    because we will output a single nifti image when we have five parts,
    but two distinct nifti images when we have nine parts.
    If the number of parts is not 5 or 9, nothing will be done.
    """
    FIVE = 5
    NINE = 9
    OTHER = None
    @classmethod
    def from_int(cls, nb_parts: int):
        for member in cls:
            if member.value == nb_parts:
                return member
        warnings.warn(
            f"Unexpected number of parts {nb_parts}. "
            f"Should be one of {[c.value for c in cls]}."
        )
        return cls.OTHER
```

If the number of parts is not 5 or 9, I'm wondering what is the best approach. We discussed giving a warning to the user (which is implemented above), but I'm a little bit afraid that the user would get flooded with warnings if there are a lot of such rows in the dataframe. We can try like this if this situation is unexpected, otherwise just remove the warning. 

```python
multipart_id = _get_multipart_id(
    PhillipsNumberOfParts.from_int(int(number_of_parts)), run_num
) 
if multipart_id is not None:
    data["MultipartID"] = multipart_id
    json.dump(data, output_file, indent=4)

def _get_multipart_id(nb_parts: PhillipsNumberOfParts, run_number: str) -> Optional[str]:
    """Return the MultiPartID for the provided run number depending on the number of parts."""
    if nb_parts == PhillipsNumberOfParts.NINE:
        if run_num in (f"run-0{k}" for k in range(1, 5)):
            return "dwi_2"
        if run_num in (f"run-0{k}" for k in range(5, 10)):
            return "dwi_1"
        raise ValueError("bla bla")
    if nb_parts == PhillipsNumberOfParts.FIVE:
        if run_num in (f"run-0{k}" for k in range(1, 5)):
            return "dwi_1"
       raise ValueError("bla bla")
    if nb_parts == PhillipsNumberOfParts.OTHER:
        return None
```

Reply:
Done !